p2p/src/peer.rs

```diff
@@ -187,10 +213,12 @@ mod run {
                         if post_receiver_len > 100 {
                             iroha_logger::warn!(size=post_receiver_len, "Peer post messages are pilling up");
                         }
-                        if let Err(error) = message_sender.send_message(msg).await {
+                        if let Err(error) = message_sender.send_message(Message::Message(msg)).await {
                             iroha_logger::error!(%error, "Failed to send message to peer.");
                             break;
                         }
+                        // Reset ping timeout as peer send another message
+                        ping_interval.reset();
```

Comment:
Does it _always_ make sense to reset the ping interval when sending a message?

AFAIU, the logic is that if we are gonna send some message, the peer's gonna respond and reset our `idle_timer`. But this assumption doesn't hold if there are messages that don't require answering..

IMO it makes more sense to reset the interval at at the same time as we reset the `idle_interval`: when a message is received

Reply:
Logic is that we don’t care if other peer going to respond to our ping, so it’s just a way to send message once in a while to remind about itself.

So these two processes are independent (reminding about ourselves and checking if other side is alive).

That’s why i reset ping interval when any message is sent.

Now i noticed the problem:
Two peers can have different idle timeouts (like 60s and 10s). 
First peer going to send pings once every 30s which is not enough to keep alive connection with the other peer who expects message every 10s.

Not sure what best course of action would be here:
1. hardcoding some value seems limiting
2. we can send desired keep alive value after the handshake to the other side (like “please send me pings once every n seconds”)
3. make it ping-pong like so that other side should reply to our ping

Reply:
Out of 2 and 3 I prefer the 3, as it would just easier to implement